### DISCUSSION TITLE
Hint on scrolling with the mouse in the VTE when `less` is used as a pager for GIT commands

### DISCUSSION THREAD

**Discussion by ghost** [geany/geany#4053] [Q&A]
Hi.

I've been having a slight issue that I haven't sorted out yet: I mostly use `git` from the VTE (for some reasons that are not relevant to the issue). Mouse wheel scrolling doesn't seem to work the way I expect it to with `git` commands when run from the VTE __and__ the pager (`less`) is involved, i.e. the _window_ is scrolled instead of the _text_. What I expect is the text to scroll with the wheel, as if I had pressed the UP or DOWN cursor keys.

I'm using this command as the shell for the VTE: `/bin/bash --rcfile ~/.vterc`

File `.vterc` is a custom script I wrote to load a minimalist configuration for `bash`.

Now here's what I've noticed after investigating:

* scrolling works as expected in a terminal window, both under `bash` and `zsh`
* scrolling works as expected when I run `less <long file>` from the VTE
* scrolling works as expected when viewing `man` pages from the VTE
* scrolling works as expected when I load `zsh` from the VTE and then type GIT commands that involve the pager (aka `less`)
* scrolling works as expected when I run `git [...] | less --raw` in the VTE... except _I loose colouring_
* scrolling does **not** work as expected when I configure the pager as `less -FX`
* scrolling does **not** work as expected when I configure the pager as `less --raw`
* scrolling does **not** work as expected when I load a new instance of `bash` in the VTE
* scrolling does **not** work as expected when I use `/bin/bash` as the VTE shell instead of the above command, which rules out my custom RC script as causing the issue

So the issue seems to be a combination between Geany AND the VTE AND Bash AND Git. I don't know where to lookup next. Do you have any hint?

P.S. : I'm running Manjaro Linux (Xfce) and Geany 2.0 (built 2023-10-23)

Here's my custom startup RC file (aka `~/.vterc`):
```sh
# This file is sourced by Geany terminal (VTE). Only a subset of the
# full profile is loaded.

# check the window size after each command and, if necessary,
# update the values of LINES and COLUMNS.
shopt -s checkwinsize

# Run a selection of common profile scripts
. /etc/profile.d/aliases.sh
. /etc/profile.d/gcc-colors.sh
. /etc/profile.d/histrc.sh

# Include this profile's aliases
[ -f "$HOME/.sh_aliases" ] && . $HOME/.sh_aliases

# To be run only under bash
if test "$BASH"; then
	# Enable colourful terminal windows
	. /etc/profile.d/prompt.sh

	# Enable GIT completion
	. /usr/share/bash-completion/completions/git

	# Redirect history when running in Geany's terminal. Also avoid the
	# ominous message "expect trouble" about screen resolution...
	if ps --no-headers -p $PPID -o comm 2>/dev/null | grep -q geany; then
		HISTIGNORE="$HISTIGNORE:*geany_run_script*"
		HISTFILE="$HOME/.cache/geany/history"
		return
	fi
fi
```

**Comment by b4n** [comment]
Doesn't the same happen in any other terminal? AFAIK it's just some less options that trigger this different behavior of the scroll wheel.

**Comment by ghost** [comment]
Nope, in other terminal windows, it's fine, with both `bash` and `zsh` — ~~and I realise I forgot to add that in my list~~ :D

I initially didn't believe it would be a bug in Geany but now I'm not so sure. Shall I make this a bug report instead?

**Reply by b4n** [reply]
Which terminal apps are you using? I'll see to test this, but the Git output never behaved like a "proper" less in terminals for me, but admittedly I only really use VTE-based terminals. I always assumed it was a less limitation with the options used to render the formatted Git output, and avoiding clearing the screen when the output was shorter than the window,  but maybe there's more to it.

**Reply by ghost** [reply]
(~~Crap, I forgot to mention I'm using Xfce~~)

The terminal app is Xfce Terminal Emulator (aka `xfce-terminal`).

**Reply by b4n** [reply]
Hum, that should use VTE as well I believe… I'll try and test this when I get to a computer, but at any rate Geany doesn't do much so at best it's a VTE option Geany doesn't set as it should or could.

**Reply by ghost** [reply]
FTR I have another machine I have recently (a few days ago) installed with Linux Mint 21.3 (Virginia) and the stock Geany with the stock GIT behaves exactly the same way in the VTE.

(By "stock" I mean the default parameters)
